### PR TITLE
[fix](csv-reader) fix bug that csv reader can not read text format hms table

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem summary

1. Missing field and line delimiter
2. When query external table with text(csv) format, we should pass the column position map to BE,
    otherwise the column order is wrong.

TODO:
1. For now, if we query csv file with non-exist column, it will return null.
    But it should return null or default value of that column.
2. Add regression test after hive docker is ready.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
4. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
5. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
6. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
7. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/format/csv/csv_reader.h [hunk 1]
```diff
@@ -57,6 +58,16 @@ class CsvReader : public GenericReader {
     const TFileScanRangeParams& _params;
     const TFileRangeDesc& _range;
     const std::vector<SlotDescriptor*>& _file_slot_descs;
+    // Only for query task, save the columns' index which need to be read.
+    // eg, there are 3 cols in "_file_slot_descs" named: k1, k2, k3
+    // and the corressponding pposition in file is 0, 3, 5.
```
Comment (BePPPower) [inline be/src/vec/exec/format/csv/csv_reader.h:63]:
```suggestion
    // and the corressponding position in file is 0, 3, 5.
```

File: be/src/vec/exec/format/csv/csv_reader.cpp [hunk 2]
```diff
@@ -134,10 +150,19 @@ Status CsvReader::get_next_block(Block* block, size_t* read_rows, bool* eof) {
...
-    auto columns = block->mutate_columns();
+    if (!_init_column_pos) {
+        // Save mapping from file slot desc to column in block, so that we don't need to
+        // find column in block by name every time.
+        for (int i = 0; i < _file_slot_descs.size(); ++i) {
+            size_t position = block->get_position_by_name(_file_slot_descs[i]->col_name());
+            _col_posistions.push_back(position);
+        }
+        _init_column_pos = true;
+    }
```
Comment (BePPPower) [inline be/src/vec/exec/format/csv/csv_reader.cpp:161]: Isn't `_file_slot_descs[i]` and `block->get_by_position(i)` corresponding?

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 41.29 seconds
 load time: 560 seconds
 storage size: 17154602812 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221020161852_clickbench_pr_32097.html

File: be/src/vec/exec/format/csv/csv_reader.cpp [hunk 3]
```diff
@@ -150,16 +175,15 @@ Status CsvReader::get_next_block(Block* block, size_t* read_rows, bool* eof) {
...
-        // TODO(ftw): check read_rows?
-        ++(*read_rows);
-        RETURN_IF_ERROR(_fill_dest_columns(Slice(ptr, size), columns));
+        RETURN_IF_ERROR(_fill_dest_columns(Slice(ptr, size), block));
+        ++rows;
 
         if (_line_reader_eof == true) {
             *eof = true;
             break;
         }
```
[the remark this answers is not in the document]
Comment (morningman) [reply be/src/vec/exec/format/csv/csv_reader.cpp]: removed

File: be/src/vec/exec/format/csv/csv_reader.cpp [hunk 4]
```diff
@@ -134,10 +150,19 @@ Status CsvReader::get_next_block(Block* block, size_t* read_rows, bool* eof) {
...
+            size_t position = block->get_position_by_name(_file_slot_descs[i]->col_name());
+            _col_posistions.push_back(position);
+        }
+        _init_column_pos = true;
+    }
 
-    while (columns[0]->size() < batch_size && !_line_reader_eof) {
+    const int batch_size = _state->batch_size();
+    int rows = 0;
+    while (rows < batch_size && !_line_reader_eof) {
```
[the remark this answers is not in the document]
Comment (morningman) [reply be/src/vec/exec/format/csv/csv_reader.cpp:155]: fixed

File: be/src/vec/exec/format/csv/csv_reader.cpp (hunk 2 above)
Comment (morningman) [reply be/src/vec/exec/format/csv/csv_reader.cpp]: removed

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 41.81 seconds
 load time: 629 seconds
 storage size: 17154816178 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221021035603_clickbench_pr_32306.html

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**